File: app/src/main/java/djordjeh/architecture/mvp/data/source/local/AppSQLiteHelper.java
package djordjeh.architecture.mvp.data.source.local;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.db.SupportSQLiteOpenHelper;
import android.provider.BaseColumns;

public class AppSQLiteHelper extends SupportSQLiteOpenHelper.Callback {

    private static final String INTEGER_TYPE = " INTEGER";
    private static final String TEXT_TYPE = " TEXT";
    private static final String COMMA_SEP = ", ";

    static class TaskEntity implements BaseColumns {
        static final String TABLE_NAME       = "tasks";
        static final String TITLE            = "title";
        static final String DESCRIPTION      = "description";
        static final String COMPLETED        = "completed";
    }

    private static final String CREATE_TABLE_TASKS =
            "CREATE TABLE " + TaskEntity.TABLE_NAME + " (" +
                    TaskEntity._ID + INTEGER_TYPE + " PRIMARY KEY" + COMMA_SEP +
                    TaskEntity.TITLE + TEXT_TYPE + COMMA_SEP +
                    TaskEntity.DESCRIPTION + TEXT_TYPE + COMMA_SEP +
                    TaskEntity.COMPLETED + INTEGER_TYPE + " );";

    /**
     * Creates a new Callback to get database lifecycle events.
     *
     * @param dbVersion The version for the database instance. See {@link #version}.
     */
    AppSQLiteHelper(int dbVersion) {
        super(dbVersion);
    }

    @Override
    public void onCreate(SupportSQLiteDatabase db) {
        db.execSQL(CREATE_TABLE_TASKS);
    }

    @Override
    public void onUpgrade(SupportSQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TaskEntity.TABLE_NAME + ";");
        onCreate(db);
    }
}

File: app/src/main/java/djordjeh/architecture/mvp/data/source/local/AppSQLiteHelper.kt
package djordjeh.architecture.mvp.data.source.local

import androidx.sqlite.db.SupportSQLiteDatabase
import androidx.sqlite.db.SupportSQLiteOpenHelper

class AppSQLiteHelper
/**
 * Creates a new Callback to get database lifecycle events.
 *
 * @param dbVersion The version for the database instance. See [.version].
 */
internal constructor(dbVersion: Int) : SupportSQLiteOpenHelper.Callback(dbVersion) {

    override fun onCreate(db: SupportSQLiteDatabase) {
        db.execSQL(CREATE_TABLE_TASKS)
    }

    override fun onUpgrade(db: SupportSQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + TaskEntity.TABLE_NAME + ";")
        onCreate(db)
    }

    internal class TaskEntity {
        companion object {
            const val ID = "_id"
            const val TABLE_NAME = "tasks"
            const val TITLE = "title"
            const val DESCRIPTION = "description"
            const val COMPLETED = "completed"
        }
    }

    companion object {

        private const val INTEGER_TYPE = " INTEGER"
        private const val TEXT_TYPE = " TEXT"
        private const val COMMA_SEP = ", "

        private const val CREATE_TABLE_TASKS = "CREATE TABLE " + TaskEntity.TABLE_NAME + " (" +
                TaskEntity.ID + INTEGER_TYPE + " PRIMARY KEY" + COMMA_SEP +
                TaskEntity.TITLE + TEXT_TYPE + COMMA_SEP +
                TaskEntity.DESCRIPTION + TEXT_TYPE + COMMA_SEP +
                TaskEntity.COMPLETED + INTEGER_TYPE + " );"
    }
}
